components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/dao/RoleDAOImpl.java

```diff
@@ -157,6 +162,11 @@ public RoleBasicInfo addRole(String roleName, List<String> userList, List<String
                         }
                         processBatchUpdateForUsers(roleName, userNamesList, tenantId, primaryDomainName, connection,
                                 addUsersSQL);
+
+                        for (String username : userNamesList) {
+                            clearUserRolesCache(username, tenantId);
+                        }
+                        clearUserRolesCacheByTenant(tenantId);
```

Comment:
If we clear the cache by tenant why do we need to clear for specific users again?

Reply:
fixed